Session-TimeOut

bbare opened the pull request:
Implemented SessionTimeOut login in the ValidateAndUpdateSession method.

# Issue/Ticket

PROGRESS ON #13 

# Description

Logic to check if a current session is valid or not. If valid, return the updated session. If not, delete the current session.

## Type of change

List one of the following: Feature

Label your PR accordingly.

# How Has This Been Tested?

Tests to come. 

# Notes

bbare committed:
TimeOut Login

Implemented SessionTimeOut login in the ValidateAndUpdateSession method.

bbare committed:
Added Delete

Deletes the current session if it is invalid.

alfredovargasjr commented on Backend/ManagerLayer/AuthorizationManagement/AuthorizationManager.cs:
```diff
@@ -37,18 +37,20 @@ public Session CreateSession(DatabaseContext _db, User user)
+			//For the Compare method, a -1 is returned if the current Session expiration is shorter than the current time
+			//A 0 is returned if they are equal
+			//A 1 is returned if current Session is longer than the current time
+			if (TimeSpan.Compare(response.ExpiresAt.TimeOfDay, DateTime.UtcNow.TimeOfDay) == -1 || TimeSpan.Compare(response.ExpiresAt.TimeOfDay, DateTime.UtcNow.TimeOfDay) == 0)
```
what about the case of someone starting the session at 11:50pm? and the check it done after midnight? just wondering

bbare replied:
I will look into it. Im pretty sure the TimeSpan.Compare method will take care of it. Will update when I have an answer.

alfredovargasjr commented on Backend/UnitTesting/UnitTesting.csproj:
```diff
@@ -60,11 +60,13 @@
     <Reference Include="System" />
     <Reference Include="System.Core" />
     <Reference Include="System.Data" />
+    <Reference Include="System.Data.Entity" />
```
was this added to fix build errors?

bbare replied:
Yes.

bbare committed:
Update AuthorizationManager.cs

bbare committed:
Added Unit Test and Fixed Code

Added a unit test for SessionTimeOut and fixed some code that had wrong logic.

bbare committed:
Added Tests and Fixs

Added more Tests for SessionTimeOut, specifically testing that a Session is deleted if it is Invalid and that a Valid Session passes. Also fixed an issue where Sessions were not being deleted.

alfredovargasjr commented on Backend/DataAccessLayer/Database/dbContext.cs:
```diff
@@ -14,12 +14,12 @@ namespace DataAccessLayer.Database
     public class DatabaseContext : DbContext
     {
 
-        public DatabaseContext()
```
Do not commit this

alfredovargasjr commented on Backend/DataAccessLayer/Models/Session.cs:
```diff
@@ -14,6 +14,13 @@ public Session()
             ExpiresAt = DateTime.UtcNow.AddMinutes(MINUTES_UNTIL_EXPIRATION);
             Id = Guid.NewGuid();
         }
+		public Session(DateTime CreatedAt, DateTime UpdatedAt, DateTime ExpiresAt)
```
dont think you need this overloaded constructor, you can just set the values via accessor

bbare committed:
SessionTimeOut

Finished SessionTimeOut

alfredovargasjr committed:
updated with master

alfredovargasjr committed:
Update dbContext.cs
